File: project/core/lib/dataset/pruner/src/main/java/vdi/core/pruner/DeletionContext.java
package vdi.core.pruner;

import org.slf4j.LoggerFactory;
import vdi.core.db.cache.model.DeletedDataset;
import vdi.logging.LoggingExt;
import vdi.logging.MarkedLogger;
import vdi.model.meta.DatasetType;
import vdi.model.meta.UserID;

import java.util.List;

public class DeletionContext {
  private final DeletedDataset dataset;

  private final MarkedLogger logger;

  private PrunableState state;

  public DeletionContext(DeletedDataset dataset) {
    this.dataset = dataset;
    this.logger = new MarkedLogger(
      new String[]{
        LoggingExt.markString(LoggingExt.PrefixUserID, dataset.getOwnerID()),
        LoggingExt.markString(LoggingExt.PrefixDatasetID, dataset.getDatasetIdString()),
      },
      LoggerFactory.getLogger(getClass())
    );
  }

  public DeletedDataset getDataset() {
    return this.dataset;
  }

  public MarkedLogger getLogger() {
    return this.logger;
  }

  public PrunableState getState() {
    return this.state;
  }

  public void setState(PrunableState state) {
    this.state = state;
  }

  public UserID getOwnerId() {
    return this.dataset.getOwnerID();
  }

  public String getDatasetId() {
    return this.dataset.getDatasetIdString();
  }

  public List<String> getProjects() {
    return this.dataset.getProjects();
  }

  public DatasetType getDataType() {
    return this.dataset.getDataType();
  }
}

File: project/core/lib/dataset/pruner/src/main/kotlin/vdi/core/pruner/DeletionContext.kt
package vdi.core.pruner

import vdi.core.db.cache.model.DeletedDataset
import vdi.logging.markedLogger

internal data class DeletionContext(val dataset: DeletedDataset) {
  val logger = markedLogger(ownerID = dataset.ownerID, datasetID = dataset.datasetID)

  lateinit var state: PrunableState

  inline val datasetID
    get() = dataset.datasetID

  inline val ownerID
    get() = dataset.ownerID

  inline val projects
    get() = dataset.projects

  inline val dataType
    get() = dataset.dataType
}
